Observer/SalesQuoteProductAddAfter.php

```diff
@@ -111,13 +125,17 @@ public function klBuildCartData($quote, $addedItem)
 
         foreach ($cartItems as $item) {
             $product = $item->getProduct();
+            # try to grab the simple product
+            $simpleProduct = $this->getSimpleProductForEvent($item);
+
             $cartItemId = $product->getId();
             $itemCategories = $product->getCategoryIds();
             $itemName = $item->getName();
             $currentProduct = [
                 'Categories' => (array) $itemCategories,
-                'ImageUrlKey' => (string) is_null($product->getData('small_image')) ? "" : stripslashes($product->getData('small_image')),
+                'ImageUrlKey' => $this->getImagePreferringVariant($product, $simpleProduct),
                 'ProductId' => (int) $cartItemId,
+                'SimpleProductId' => (int) is_null($simpleProduct) ? null : $simpleProduct->getId(),
```

Comment:
Just confirming if this is doing what we want. If `$simpleProduct` is null are we hoping to return `null` or `0` here? I don't believe we need to cast the conditional expression and that appears to be what's happening here.